beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -631,19 +621,16 @@ impl<T: BeaconChainTypes> DataAvailabilityCheckerInner<T> {
         };
 
         // If we're sampling all columns, it means we must be custodying all columns.
-        let custody_column_count = self.sampling_column_count();
         let total_column_count = self.spec.number_of_columns as usize;
         let received_column_count = pending_components.verified_data_columns.len();
 
         if pending_components.reconstruction_started {
             return ReconstructColumnsDecision::No("already started");
         }
-        if custody_column_count != total_column_count {
-            return ReconstructColumnsDecision::No("not required for full node");
-        }
         if received_column_count >= total_column_count {
             return ReconstructColumnsDecision::No("all columns received");
         }
+        // Only supernodes receive >= 50% of columns
```

Comment:
Removed the "not required for full node" condition as it's redundant and would force us to pipe the custody group here somehow

Reply:
I had a quick look and didn't see any existing check to ensure this wasn't being called on a non-full node. Why do you say it's redundant?

Could you not do this check with:
```rust
if let Some(custody_columns_count) = pending_components
    .executed_block
    .as_ref()
    .map(|diet_block| diet_block.custody_columns_count())
{
    if custody_columns_count != total_column_count {
        return ReconstructColumnsDecision::No("not required for full node");
    }
}
```

Reply:
What you suggest will cause weird behavior:
- if block is there: reconstruction happens only if custody_columns_count = 100%
- if block is not there: reconstruction happens only if custody_columns_count = 50%

Block being there or not before attempting to reconstruction depends on network conditions and timings.

Reply:
Are nodes that custody 50% of columns considered supernodes?
I'm not sure if there's any reason to  reconstruct for these nodes, as they are not required to re-publish to their non-custody columns.

> If the node is subscribed to the subnet corresponding to the column, it MUST send the reconstructed DataColumnSidecar to its topic mesh neighbors. If instead the node is not subscribed to the corresponding subnet, it SHOULD still expose the availability of the DataColumnSidecar as part of the gossip emission process.

I don't quite understand why it means to "expose the availability of the DataColumnSidecar as part of the gossip emission process"

https://github.com/ethereum/consensus-specs/blob/dev/specs/fulu/das-core.md#reconstruction-and-cross-seeding

Reply:
Also, wouldn't this change cause this error since we're sending 128 columns to the  DA checker upon successful reconstruction:
> "too many columns got {num_received_columns} expected {num_expected_columns}"

If we do end up reconstructing for these nodes that don't custody 100% of columns, we would have to filter before putting them into the DA checker.

Reply:
> Removed the "not required for full node" condition as it's redundant and would force us to pipe the custody group here somehow

Piping this through should be fairly straight forward as reconstruction are called from `NetworkBeaconProcessor` and has access to `globals`

https://github.com/sigp/lighthouse/blob/187467de28b37c5b933b8aba046c01a9aa32e839/beacon_node/network/src/network_beacon_processor/mod.rs#L1008-L1012

Reply:
@dapplion made a fix here to ensure this isn't called on a non supernode
https://github.com/sigp/lighthouse/pull/7033/commits/62660577f7dd8f326ba906ef5e003584c29453dc